PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -344,6 +344,10 @@ protected function do_select($sec, $usec)
         $write = null;
         $except = null;
 
+        if ($sec === null && PHP_VERSION_ID >= 80100) {
+            $usec = null;
+        }
+
         return stream_select($read, $write, $except, $sec, $usec);
```

Comment:
Failed test: https://github.com/javer/php-amqplib/runs/3210161338?check_suite_focus=true#step:11:55
```
ErrorException: stream_select(): Argument #5 ($microseconds) should be null instead of 0 when argument #4 ($seconds) is null

/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php:347
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/AbstractIO.php:88
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php:129
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php:155
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php:103
...
```

Since PHP 8.1 signature of the function `stream_select` has been changed: https://github.com/php/php-src/blob/master/ext/standard/basic_functions.stub.php#L1215
```php
function stream_select(?array &$read, ?array &$write, ?array &$except, ?int $seconds, ?int $microseconds = null): int|false {}
```

Reply:
I would suggest to fix this for all versions. It makes no sense to use microseconds with non null seconds.

Reply:
Unfortunately it won't work, because previous versions of PHP don't accept null for `microseconds` argument, see https://github.com/php/php-src/blob/php-8.0.9/ext/standard/basic_functions.stub.php#L1198
```php
function stream_select(?array &$read, ?array &$write, ?array &$except, ?int $seconds, int $microseconds = 0): int|false {}
```